assets/src/components/searchForm.tsx

```diff
@@ -78,17 +83,22 @@ type SearchFormProps = SearchFormEventProps &
      * Callback to run when {@link inputText} should be updated.
      */
     onInputTextChange?: React.ChangeEventHandler<HTMLInputElement>
-
-    /**
-     * The property being searched
-     */
-    property: SearchPropertyQuery
-    /**
-     * Callback to run when {@link property} should be updated.
-     */
-    onPropertyChange: (property: SearchPropertyQuery) => void
   }
 
+type SearchComboboxProps = ComboboxProps & {
+  comboboxType: "map_search"
+  dispatch: Dispatch
+  query: SearchQuery
+  options: null
+}
+
+type SelectComboboxProps = ComboboxProps & {
+  comboboxType: "select"
+  dispatch: null
+  query: null
```

Comment:
nit/suggestion(non-blocking): I _think_ if you make these optional, then they don't need to be specified when the `comboboxType` is `select`
Which would also mean if we're able to change the interface at some point, we wouldn't need to track these props down at that point

same for `options` on `SearchComboboxProps`

```suggestion
  dispatch?: undefined
  query?: undefined
```

Reply:
Oh my goodness, there ya go! I tried a couple typing adjustments but not this. Thank you 🙇